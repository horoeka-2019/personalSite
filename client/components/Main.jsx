import React from "react";
import { Link, HashRouter as Router } from "react-router-dom";

class Main extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Router>
          <div className="container">
            <div className="column-1">
              <Link to="/blog">
                <div className="title">Blog</div>
                <p className="intro">Technical and learning reflections</p>
                <div className="image">
                  <img
                    src="images/blogfancy.svg"
                    alt="blog icon"
                    height="140"
                  />
                </div>
              </Link>
            </div>
            <div className="column-2">
             <Link to="/work">
              <div className="title">Work</div>
                <p className="intro">Examples of projects and tinkerings</p>
                  <div className="image">
                    <img
                      src="images/workfancy.svg"
                      alt="blog icon"
                      height="140"
                    />
                 </div>
             </Link>
            </div>
            <div className="column-3">
              <Link to="/about">
                <div className="title">About</div>
                <p className="intro">Contact details and a little about me</p>
                  <div className="image">
                    <img
                      src="images/aboutfancy.svg"
                      alt="blog icon"
                      height="140"
                    />
                  </div>
              </Link>
            </div>
            </div>
        </Router>
      </React.Fragment>
    )
  }
}

export default Main

