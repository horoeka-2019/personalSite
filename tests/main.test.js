import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Main from '../client/components/Main'

describe('<Main /> component test', () => {
  it('renders work title', () => {
    // Arrange
    const expected = 'Work'

    // Act
    const wrapper = mount(<Main />)
    const actual = wrapper.text()
    // Assert
    expect(actual).toContain(expected)
  })

  it('renders About title', () => {
    // Arrange
    const expected = 'About'

    // Act
    const wrapper = mount(<Main />)
    const actual = wrapper.text()
    // Assert
    expect(actual).toContain(expected)
  })
  it('render unchanged from previous snapshot', () => {
    const wrapper = mount(<Main/>)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
 
})