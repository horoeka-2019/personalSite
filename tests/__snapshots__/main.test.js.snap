// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component test render unchanged from previous snapshot 1`] = `
<Main>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div
        className="container"
      >
        <Link
          to="/work"
        >
          <LinkAnchor
            href="#/work"
            navigate={[Function]}
          >
            <a
              href="#/work"
              onClick={[Function]}
            >
              <div
                className="column-1"
              >
                <div
                  className="title"
                >
                  Work
                </div>
                <p
                  className="intro"
                >
                  Examples of projects and tinkerings
                </p>
                <div
                  className="image"
                >
                  <img
                    alt="work icon"
                    height="120"
                    src="images/workfancy.svg"
                  />
                </div>
              </div>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          to="/about"
        >
          <LinkAnchor
            href="#/about"
            navigate={[Function]}
          >
            <a
              href="#/about"
              onClick={[Function]}
            >
              <div
                className="column-2"
              >
                <div
                  className="title"
                >
                  About
                </div>
                <p
                  className="intro"
                >
                  Contact details and a little about me
                </p>
                <div
                  className="image"
                >
                  <img
                    alt="about icon"
                    height="120"
                    src="images/aboutfancy.svg"
                  />
                </div>
              </div>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </Router>
  </HashRouter>
</Main>
`;
